httpx/_models.py

```diff
@@ -401,6 +401,72 @@ class Headers(typing.MutableMapping[str, str]):
     HTTP headers, as a case-insensitive multi-dict.
     """
 
+    class _HeadersKeysView(typing.KeysView[str]):
+        def __init__(
+            self, raw: typing.List[typing.Tuple[bytes, bytes]], encoding: str
+        ) -> None:
+            self._raw = raw
+            self._encoding = encoding
+
+        def __contains__(self, key: typing.Any) -> bool:
+            try:
+                normalized_key = key.lower().encode(self._encoding)
+            except AttributeError:
+                return False
+            return any(k == normalized_key for k, _ in self._raw)
+
+        def __iter__(self) -> typing.Iterator[str]:
+            return iter(k.decode(self._encoding) for k, _ in self._raw)
+
+        def __len__(self) -> int:
+            return len(self._raw)
+
+    class _HeadersValuesView(typing.ValuesView[str]):
+        def __init__(
+            self, raw: typing.List[typing.Tuple[bytes, bytes]], encoding: str
+        ) -> None:
+            self._raw = raw
+            self._encoding = encoding
+
+        def __contains__(self, value: typing.Any) -> bool:
+            return any(v.decode(self._encoding) == value for _, v in self._raw)
+
+        def __iter__(self) -> typing.Iterator[str]:
+            return iter(v.decode(self._encoding) for _, v in self._raw)
+
+        def __len__(self) -> int:
+            return len(self._raw)
+
+    class _HeadersItemsView(typing.ItemsView[str, str]):
+        def __init__(
+            self, raw: typing.List[typing.Tuple[bytes, bytes]], encoding: str
+        ) -> None:
+            self._raw = raw
+            self._encoding = encoding
+
+        def __contains__(self, item: typing.Any) -> bool:
+            try:
+                key, value = item
+            except (TypeError, ValueError):
+                return False
+
+            try:
+                normalized_key = key.lower().encode(self._encoding)
+                normalized_value = value.encode(self._encoding)
```

Comment:
is there a compelling reason not to use `normalize_header_key`, `normalize_header_value` from [`_utils.py`](https://github.com/encode/httpx/blob/7ab7efd5b04c00a4449c4ee89aea832286919666/httpx/_utils.py#L34) here?